### PR TITLE
Remove Jenkins from pages

### PR DESCRIPTION
**Pull request by bjornjorgensen**
`bundle exec jekyll build --watch`

Removed info about Jerkins from the developer-tools.md fil 

      Regenerating: 1 file(s) changed at 2023-02-25 15:14:42
                    developer-tools.md
                    ...done in 2.778396909 seconds.

We don't use Jenkins anymore so we can remove this info.

### CODE REVIEW

**Review by srowen — APPROVED** [review]

**Comment by bjornjorgensen** [comment]
hmm.. there is one more 

"If you open a pull request containing binary incompatibilities anyway, Jenkins will remind you by failing the test build with the following message:

Test build #xx has finished for PR yy at commit ffffff.

  This patch fails MiMa tests.
  This patch merges cleanly.
  This patch adds no public classes."

should I remove this one to?

**Comment by bjornjorgensen** [comment]
And in https://spark.apache.org/release-process.html 

"It is also good to set up Jenkins jobs for the release branch once it is cut to ensure tests are passing. These are jobs like https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test/job/spark-branch-2.3-test-maven-hadoop-2.7/ . Consult Josh Rosen and Shane Knapp for help with this. Also remember to add the newly-added jobs to the test dashboard at https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test%20(Dashboard)/ ."


"Also check that all build and test passes are green from the RISELab Jenkins: https://amplab.cs.berkeley.edu/jenkins/ particularly look for Spark Packaging, QA Compile, QA Test. Note that not all permutations are run on PR therefore it is important to check Jenkins runs."

**Comment by bjornjorgensen** [comment]
and in https://spark.apache.org/contributing.html  

"The Jenkins automatic pull request builder will test your changes
If it is your first contribution, Jenkins will wait for confirmation before building your code and post “Can one of the admins verify this patch?”
A committer can authorize testing with a comment like “ok to test”
A committer can automatically allow future pull requests from a contributor to be tested with a comment like “Jenkins, add to whitelist”
After about 2 hours, Jenkins will post the results of the test to the pull request, along with a link to the full results on Jenkins.
Watch for the results, and investigate and fix failures promptly
Fixes can simply be pushed to the same branch from which you opened your pull request
Jenkins will automatically re-test when new commits are pushed
If the tests failed for reasons unrelated to the change (e.g. Jenkins outage), then a committer can request a re-test with “Jenkins, retest this please”. Ask if you need a test restarted. If you were added by “Jenkins, add to whitelist” from a committer before, you can also request the re-test."

**Comment by bjornjorgensen** [comment]
Now there are something about running jenkins test on github enterprise that are left I hope.  https://spark.apache.org/docs/latest/building-spark.html#running-jenkins-tests-with-github-enterprise

Where should I remove this one then?

File: contributing.md
Comment (bjornjorgensen) [inline contributing.md:390]: Here we can write about enable github actions on fork. And that github actions are running test.

File: developer-tools.md
Comment (bjornjorgensen) [inline developer-tools.md:277]: What does Github actions do her?

Comment (srowen) [reply developer-tools.md:277]: I would just remove these references, all of the Jenkins references. I think it's OK, rather than rewriting them in terms of GHA. We can add back docs later

**Comment by bjornjorgensen** [comment]
@dongjoon-hyun FYI

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. @bjornjorgensen and @srowen

**Review by viirya — APPROVED** [review]

**Pull request closed without merging**
